src/createServer.js

```diff
@@ -1,3 +1,34 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+const { handleErrors } = require('./handleErrors');
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    const normalizedURL = new URL(req.url, `http://${req.headers.host}`);
+    const textToConvert = normalizedURL.pathname.slice(1);
+    const caseName = normalizedURL.searchParams.get('toCase');
+
+    res.setHeader('Content-Type', 'application/json');
+
+    const errors = handleErrors(textToConvert, caseName);
+
+    if (errors.length) {
+      res.statusCode = 400;
+      res.statusMessage = 'Bad request';
+
+      res.end(JSON.stringify({ errors }));
+    } else {
+      const conversionResult = convertToCase(textToConvert, caseName);
+
+      conversionResult.targetCase = caseName;
+      conversionResult.originalText = textToConvert;
+
+      res.statusCode = 200;
+      res.end(JSON.stringify(conversionResult));
+    }
```

Comment:
Is `else` necessary here?
You have a case when you have the errors and default case.

 ```suggestion
    }
    const conversionResult = convertToCase(textToConvert, caseName);

    conversionResult.targetCase = caseName;
    conversionResult.originalText = textToConvert;

    res.statusCode = 200;
    res.end(JSON.stringify(conversionResult));
```